src/receive_imf.rs

```diff
@@ -826,9 +815,9 @@ async fn add_parts(
                         // The message itself will be sorted under the device message since the device
                         // message is `MessageState::InNoticed`, which means that all following
                         // messages are sorted under it.
-                        let sort_timestamp =
-                            calc_sort_timestamp(context, sent_timestamp, chat_id, true, incoming)
-                                .await?;
+                        let sort_timestamp = chat_id
+                            .calc_protection_msg_sort_timestamp(context, mime_parser.timestamp_sent)
+                            .await?;
                         chat_id
                             .set_protection(context, new_protection, sort_timestamp, Some(from_id))
                             .await?;
```

Comment:
Nit, feel free to ignore:

We're almost always calling `calc_protection_msg_sort_timestamp()` right before `set_protection()` - except for `create_group_chat()`, where it wouldn't hurt (and one database request whenever a group is created isn't a lot - what we should avoid is more database requests every time a message is received).

So, `calc_protection_msg_sort_timestamp()` _could_ be moved into `set_protection()`. And since it would then be used only a single time, it could be inlined. Both these steps would reduce code complexity I think.

Reply:
Added `ChatId::set_protection_for_timestamp_sent()` for this. Better ideas for the function name are welcome

Reply:
Another option is to add an enum for timestamps with variants `Sent(i64), Sort(i64)`, but not sure it's better

Reply:
For the name, you could rename `set_protection()` to `set_protection_for_timestamp_sort()` since it's an internal function now that's only used within `chat.rs` and doesn't need a nice name. And rename `set_protection_for_timestamp_sent()` to `set_protection()` since it's the function that everyone should use when in doubt.

In any case, you should inline `calc_protection_msg_sort_timestamp()` since it's a two-line function that is used only a single time.

Reply:
Done